src/transaction/log_recovery.c

```diff
@@ -1379,9 +1380,42 @@ log_rv_analysis_complete (THREAD_ENTRY * thread_p, int tran_id, LOG_LSA * log_ls
        * holding a page.
        */
       log_lsa->pageid = NULL_PAGEID;
-      log_recovery_resetlog (thread_p, &record_header_lsa, prev_lsa);
       *did_incom_recovery = true;
 
+      /*
+       * We can't use LOG_FIND_TDES (tran_index) here because the compelete log record
+       * may be the first log record seen for the transaction.
+       */
+      auto tdes = logtb_rv_find_allocate_tran_index (thread_p, tran_id, log_lsa);
+      if (tdes == nullptr)
+	{
+	  logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "log_rv_analysis_complete");
+	  return ER_FAILED;
+	}
+
+      if (tdes->rcv.tran_start_postpone_lsa.is_null () == false)
+	{
+	  /*
+	   * When there were postpone operations, prevent redoing them
+	   * and restore undo info starting from LOG_COMMIT_WITH_POSTPONE.
+	   */
+	  assert (tdes->state == TRAN_UNACTIVE_COMMITTED_WITH_POSTPONE);
+
+	  log_recovery_resetlog (thread_p, &tdes->rcv.tran_start_postpone_lsa, &NULL_LSA);
+
+	  auto header = (LOG_RECORD_HEADER *) ((char *) LOG_APPEND_PTR ());
```

Comment:
What happens if the LOG_COMMIT record and LOG_COMMIT_WITH_POSTPONE record exist on the different log page?

Reply:
log_recovery_resetlog() -> logpb_fetch_start_append_page() -> logpb_locate_page()

@joohok explains this.

Reply:
> @joohok explains this.

@hornetmj Does it mean no more elaboration is required?

Reply:
@wrlawodms No, @joohok teaches me enough.